manager/manager.go

```diff
@@ -783,6 +790,35 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
 	return nil
 }
 
+// Enables overwriting an existing containerData/Handler object for a given containerName.
+// can't use createContainer as it just returns if a given containerName has a handler already
+// Ex: rkt handler will want to take priority over the raw handler, but the raw handler might be created first
+
+// Only allow raw handler to be overriden
+func (m *manager) overrideContainer(containerName string, watchSource watcher.ContainerWatchSource) error {
```

Comment:
ok, so all you are asking for is removing the event, I'll try something and
see if we're on the same page

On Mon, May 16, 2016 at 3:35 PM, Tim St. Clair <notifications@github.com>
wrote:

> In manager/manager.go
> <https://github.com/google/cadvisor/pull/1284#discussion_r63437740>:
>
> > @@ -783,6 +790,35 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
> >  	return nil
> >  }
> >
> > +// Enables overwriting an existing containerData/Handler object for a given containerName.
> > +// can't use createContainer as it just returns if a given containerName has a handler already
> > +// Ex: rkt handler will want to take priority over the raw handler, but the raw handler might be created first
> > +
> > +// Only allow raw handler to be overriden
> > +func (m *manager) overrideContainer(containerName string, watchSource watcher.ContainerWatchSource) error {
>
> What do you mean add flags to the event struct? I'd prefer for the logic
> to be in the manager since it actually knows which watcher sources are
> running. The watchers should be independent. Why would the raw handler try
> to overwrite itself?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1284/files/29e0782cedec140df582c9422f4414e4cb3d611d#r63437740>
>